data/data_manager.py

```diff
@@ -87,10 +152,19 @@ def _append_csv_data(self):
                     for col in cols.drop('time'):
                         # Raise error if col already appended
                         if col in appended.keys():
-                            raise ReferenceError('{0} in multiple files within the Resources folder. These are: {1}, and {2}'.format(col, appended[col], f))
-                        # Trim df from cols that are not in categories
-                        elif not( (col in all_keys) or any(col.startswith(key) for key in zon_keys) ):
+                            raise ReferenceError('{0} in multiple files within the Resources folder.'\
+                                                 'These are: {1}, and {2}'.format(col, appended[col], f))
+                        # Trim df from cols that are not allowed
+                        elif not( (col in zon_keys) or (col in bou_keys) ):
                             df.drop(col, inplace=True)
+                            raise KeyError('The data column {0} from file {1} '\
```

Comment:
@JavierArroyoBastida Raising an error in ``load_data_and_kpisjson()`` is good since that is the data is being used in the deployed test case.  For ``_append_csv_data()``, in the past a column that did not conform to the specification was just dropped and not included in the final test case fmu.  With this implementation, an error is being raised and the process is being stopped if this is the case.  Just want to double check that this functionality is what you intend.  It is a stricter test, though somewhat less convenient if a column in a csv used for test data is derived from other columns in a csv, which may not be used for test data.  I am tending toward the stricter test, but wanted to double check with you.

Reply:
@dhblum I was indeed doubting whether a warning or an error was most convenient. In the end I decided to use the error because it avoids saving (and loading) the test case with unnecessary data. If these data needs to be saved for any reason it can still be stored in any other folder. Also if the test case developer is not using the `data_generator` but excel sheets with dependent columns, it should not be an issue to filter out the dispensable columns, for instance when changing the format from xlsx to csv.

Reply:
Just to be clear 

> In the end I decided to use the error because it avoids saving (and loading) the test case with unnecessary data

refers to saving and loading data from the test case DIRECTORY, not the test case FMU?  Because the unused data is dropped before saving the test case FMU.

Reply:
By "saving" I refer to the process of storing the data from the test case directory into the fmu like in:
https://github.com/JavierArroyoBastida/project1-boptest/blob/4b37fbcdeff844878e10310722a0c200f0733fc9/data/data_manager.py#L183
By "loading" I refer to the process of retrieving the data from the fmu into the test case object like in:
https://github.com/JavierArroyoBastida/project1-boptest/blob/4b37fbcdeff844878e10310722a0c200f0733fc9/data/data_manager.py#L327
Both methods are implemented with errors instead of warnings to prevent both: saving unnecessary data into the test case fmu and loading unnecessary data into the test case object. The latter would be used only for compiled tests case fmus that save their data by other means than using the `data_manager.save_data_and_kpisjson` method, would there be any case like that one? I'm wondering now if it's really needed...

Reply:
Yes, a test case developer could compile their own FMU with data.

And ok about the "saving" and "loading".  My point is that ``save_data_and_kpisjson `` uses ``_append_csv_data()``.  If the "saving" method uses a warning rather than an error, than the unnecessary data would still NOT be saved in the test case FMU in the end.  It is even described by https://github.com/JavierArroyoBastida/project1-boptest/blob/4b37fbcdeff844878e10310722a0c200f0733fc9/data/data_manager.py#L170.

Reply:
You're right, I didn't notice but see your point now. Still I think it's better practice to raise an error to ensure that emulator developers know what is being saved in the test case fmu, warnings can be easily missed.

Reply:
That sounds good.  So should Line 170 - 176 and 159 be edited?

Reply:
Indeed, I'm going to push changes for that.